src/component/handlers/edit/editOnBeforeInputAndroid.js

```diff
@@ -0,0 +1,198 @@
+'use strict';
+
+import type DraftEditor from 'DraftEditor.react';
+import type { DraftInlineStyle } from 'DraftInlineStyle';
+
+const ReactDOM = require('ReactDOM');
+
+const DraftModifier = require('DraftModifier');
+const EditorState = require('EditorState');
+const getDraftEditorSelection = require('getDraftEditorSelection');
+const invariant = require('invariant');
+
+const keyCommandPlainBackspace = require('keyCommandPlainBackspace');
+const keyCommandPlainDelete = require('keyCommandPlainDelete');
+const keyCommandInsertNewline = require('keyCommandInsertNewline');
+
+const getEntityKeyForSelection = require('getEntityKeyForSelection');
+const getDraftEditorSelectionWithNodes = require('getDraftEditorSelectionWithNodes');
+
+/**
+ * Replace the current selection with the specified text string, with the
+ * inline style and entity key applied to the newly inserted text.
+ */
+function replaceText(
+  editorState: EditorState,
+  text: string,
+  inlineStyle: DraftInlineStyle,
+  entityKey: ?string,
+): EditorState {
+  var contentState = DraftModifier.replaceText(
+    editorState.getCurrentContent(),
+    editorState.getSelection(),
+    text,
+    inlineStyle,
+    entityKey,
+  );
+  return EditorState.push(editorState, contentState, 'insert-characters');
+}
+
+const logChanges = (editor, mutation) => {
```

Comment:
Need to ditch this

Reply:
ohhhh yeah will remove

Reply:
👍 Yeah, it's slow and only used for debugging.

Reply:
it is gone